google_guest_agent/network/manager/systemd_networkd_linux.go

```diff
@@ -514,34 +560,120 @@ func (n systemdNetworkd) Rollback(ctx context.Context, nics *Interfaces) error {
 		return fmt.Errorf("failed to get list of interface names: %v", err)
 	}
 
+	ethernetRequiresRestart := false
+
 	// Rollback ethernet interfaces.
 	for _, iface := range interfaces {
-		// Find expected files.
-		configFile := n.networkFile(iface)
-		sections := new(systemdConfig)
+		reqRestart, err := n.rollbackNetwork(n.networkFile(iface))
+		if err != nil {
+			logger.Infof("Failed to rollback .network file: %v", err)
+		}
 
-		logger.Debugf("checking for %s", configFile)
-		if err := readIniFile(configFile, sections); err != nil {
-			return fmt.Errorf("failed to read systemd's .network file: %+v", err)
+		if reqRestart {
+			ethernetRequiresRestart = true
 		}
 
-		// Check that the guest section exists and the key is set to true.
-		if sections.isGuestAgentManaged() {
-			logger.Debugf("removing %s", configFile)
-			if err = os.Remove(configFile); err != nil && !os.IsNotExist(err) {
-				return err
-			}
+		reqRestart, err = n.rollbackNetdev(n.networkFile(iface))
+		if err != nil {
+			logger.Infof("Failed to rollback .network file: %v", err)
+		}
+
+		if reqRestart {
```

Comment:
would it be simpler if we use separate local variables for is `reqRestart` (for e.g. reqRestart1, reqRestart2...) and eventually just check if either of them is `true`? That way we can avoid these many if checks after every step.

Reply:
done.